tests/recordCallsToSetTimeout.ts

```diff
@@ -0,0 +1,193 @@
+import EventEmitter from 'events';
+import EMPTY_FUNCTION from './emptyFunction';
+
+type OriginalSetTimeout = typeof global.setTimeout;
+
+type SetTimeoutCallback = () => any;
+
+interface SetTimeoutCall {
+  callback: SetTimeoutCallback;
+  duration: number;
+  timeout: NodeJS.Timeout;
+}
+
+type InterceptingCallback = (
+  callback: SetTimeoutCallback,
+  stopPassingThroughCalls: () => void,
+) => SetTimeoutCallback;
+
+/**
+ * A class that provides a mock implementation for `setTimeout` which records
+ * the callback given so that it can be replayed later.
+ */
+class SetTimeoutRecorder {
+  private originalSetTimeout: OriginalSetTimeout;
```

Comment:
Nit: These can be ECMA private fields

Reply:
Good point! Changed in [d9efdd6](https://github.com/MetaMask/eth-block-tracker/pull/103/commits/d9efdd607d475bf486b93d6f695f3223c94c4dc1).